# num 13

##### № 27768 Апробация 04.03.26

Сеть задана IP-адресом 172.16.160.0 и маской сети 255.255.240.0.

Сколько в этой сети IP-адресов, для которых количество единиц в их двоичной записи кратно 2?

In [ ]:
from ipaddress import *

cnt = 0
net = ip_network("172.16.160.0/255.255.240.0", 0)
for add in net:
    bin_add = bin(int(add))[2:]
    if bin_add.count("1") % 2 == 0:
        cnt += 1
print(cnt)

Ответ: 2048

##### № 27625 Апробация 04.03.26 

Сеть задана IP-адресом 172.16.96.0 и маской сети 255.255.224.0.

Сколько в этой сети IP-адресов, для которых количество единиц в их двоичной записи кратно 2?

In [ ]:
from ipaddress import *

cnt = 0
net = ip_network("172.16.96.0/255.255.224.0", 0)
for add in net:
    bin_add = bin(int(add))[2:]
    if bin_add.count("1") % 2 == 0:
        cnt += 1
print(cnt)

Ответ: 4096

##### № 25352 ЕГКР 13.12.25

Сеть задана IP-адресом одного из входящих неё узлов 190.202.83.62 и сетевой маской 255.255.252.0.

Найдите наибольший IP-адрес в данной сети, который может быть назначен компьютеру. В ответе укажите сумму числовых значений октетов найденного IP-адреса.

Например, если бы найденный адрес был равен 100.20.3.4, то в ответе следовало бы записать: 127.

In [ ]:
from ipaddress import ip_network

net = ip_network("190.202.83.62/255.255.252.0", False)
list_add = [x for x in net.hosts()]
print(sum(int(x) for x in str(list_add[-1]).split(".")))

Ответ: 729

##### № 29966 Апробация 14.05.26 (Уровень: Базовый)

Сеть задана IP-адресом одного из входящих в неё узлов 146.180.173.153 и сетевой маской 255.192.0.0. Найдите наибольший в данной сети IP-адрес, который может быть назначен компьютеру. В ответе укажите найденный IP-адрес без разделителей.
Например, если бы найденный адрес был равен 111.22.3.44, то в ответе следовало бы записать: 11122344.

In [ ]:
from ipaddress import *

net = ip_network("146.180.173.153/255.192.0.0", 0)
adds = [add for add in net.hosts()]
print(str(adds[-1]).replace(".", ""))


Ответ: 146191255254

##### № 29345 Открытый вариант 2026 (Уровень: Базовый)

Сеть задана IP-адресом одного из входящих в неё узлов 68.203.243.87 и сетевой маской 255.255.224.0.
Найдите наибольший в данной сети IP-адрес, который может быть назначен компьютеру. В ответе укажите сумму числовых значений октетов найденного IP-адреса.
Например, если бы найденный адрес был равен 100.20.3.4, то в ответе следовало бы записать: 127

In [14]:
from ipaddress import *
net = ip_network("68.203.243.87/255.255.224.0", 0)
res = [sum(int(x) for x in str(add).split(".")) for add in net.hosts()]
print(res[-1])

780


Ответ: 780

##### № 28934 ЕГКР 18.04.26 (Уровень: Базовый)

Сеть задана IP-адресом одного из входящих в неё узлов 191.89.109.206 и сетевой маской 255.255.224.0.
Определите наибольший IP-адрес данной сети, который может быть присвоен компьютеру. В ответе укажите сумму числовых значений октетов найденного IP-адреса.
Например, если бы найденный адрес был равен 100.20.3.4, то в ответе следовало бы записать 127.

In [15]:
from ipaddress import *

net = ip_network("191.89.109.206/255.255.224.0", 0)
adds = [str(add) for add in net.hosts()]
print(sum(int(x) for x in adds[-1].split(".")))

661


Ответ: 661